### DISCUSSION TITLE
What if all available packages are pre-installed along with the kernel?

### DISCUSSION THREAD

**Discussion by MRYingLEE** [jupyterlite/xeus-python-kernel#183] [Q&A]
The effect could be:
Disk: More of course or same for some CDNs are used
Memory: Same unless the pre-installed packages are used.
Building procedure: longer time

Am I right? If so, why don't we always make all technically ready packages pre-installed? Especially the WASM ones.

Any ideas?

**Comment by martinRenou** [comment]
How would you deal with package versions constraint? You'd be lucky to be able to install all available packages without having version conflicts. Also the packed environment would be super huge, unless we use a CDN, which we plan to allow in the future but it's not yet possible.

You can try, but I don't think this would be a viable default.
